.github/workflows/build-and-test.yml

```diff
@@ -13,50 +13,46 @@ jobs:
       uses: actions/checkout@v2.4.0
 
     - name: Install nix.
-      uses: cachix/install-nix-action@v15
+      uses: cachix/install-nix-action@v18
       with:
-        nix_path: nixpkgs=channel:nixos-unstable
-      # Would using cachix help in anyway? I think we are not building anything, we're
-      # only fetching things from cache.nixos.org and iohk.cachix.org
-      # - uses: cachix/cachix-action@v10
-      #   with:
-      #     name: plutus-libs
-      #     authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+        ## Some flake caching
+        extra_nix_config: |
+          access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
       
+    ## Example from
+    ## https://github.com/actions/cache/blob/ac25611caef967612169ab7e95533cf932c32270/examples.md#haskell---cabal
     - name: Accessing the cabal cache.
-      uses: actions/cache@v2
+      uses: actions/cache@v3
       with:
         path: |
+          ~/.cabal/packages
           ~/.cabal/store
           dist-newstyle
-        key: cabal-cache-0-${{ github.sha }}
-        restore-keys: |
-          cabal-cache-0-
+        key: ${{ runner.os }}-${{ hashFiles('**/*.cabal', '**/cabal.project', '**/cabal.project.freeze') }}
+        restore-keys: ${{ runner.os }}-
 
     - name: Accessing the result cache
-      uses: actions/cache@v2
+      uses: actions/cache@v3
       with:
         path: |
           ./*.out
           ./*.res
         key: result-cache-${{ github.sha }}
```

Comment:
Just like in Pirouette, this should probably be an artifact upload instead (with an artifact download later on).